modules/sqlite_helpers.py

```diff
@@ -125,7 +125,13 @@ def maybe_delete_expired_url(sqlite_file, sqlite_row) -> bool: #returns True if
     year_ago_date = datetime.now() - timedelta(days=365)
     result_datetime_str = sqlite_row[3].split(".")[0]  # Remove fractional seconds
     result_datetime = datetime.strptime(result_datetime_str, "%Y-%m-%d %H:%M:%S")
-    if result_datetime < year_ago_date:
+
+    expiration_datetime = None
+    if sqlite_row[5] is not None:
+        expiration_datetime_str = sqlite_row[5].split(".")[0]
+        expiration_datetime = datetime.strptime(expiration_datetime_str, "%Y-%m-%d %H:%M:%S")
```

Comment:
also i think we dont need string.split, we can parse the row as is, like

```suggestion
        expiration_datetime = datetime.strptime(sqlite_row[5], "%Y-%m-%d %H:%M:%S.%f")
```

let me know if this works

Reply:
yes that works